src/bin/FacilitatorStart.ts

```diff
@@ -0,0 +1,168 @@
+            facilitatorConfig,
+            originChain,
+            auxChain,
+          )
+        ) {
+          if (options.mosaicConfig === undefined) {
+            const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(originChain);
+            configObj = new Config(mosaicConfig, facilitatorConfig);
+          } else {
+            // when mosaic config, facilitator config, origin chain id and aux chain id is given.
+            // verify origin chain and aux chain is present in mosaic and facilitator config.
+            const mosaicConfig: MosaicConfig = MosaicConfig.fromFile(options.mosaicConfig);
+            FacilitatorStart.verifyChainIdInMosaicConfig(mosaicConfig, originChain, auxChain);
+            configObj = Config.getConfigFromPath(options.mosaicConfig, options.facilitatorConfig);
+          }
+        }
+      } else if (options.mosaicConfig !== undefined) {
+        const mosaic: MosaicConfig = MosaicConfig.fromFile(options.mosaicConfig);
+        if ((FacilitatorStart.verifyChainIdInMosaicConfig(mosaic, originChain, auxChain))) {
+          const facilitator = FacilitatorConfig.from(auxChain);
+          configObj = new Config(mosaic, facilitator);
+        }
+      } else {
+        // Only origin chain and aux chain id is provided.Get the configs from default path.
+        const facilitator: FacilitatorConfig = FacilitatorConfig.from(auxChain);
+        const mosaic: MosaicConfig = MosaicConfig.fromChain(originChain);
+        configObj = new Config(mosaic, facilitator);
+      }
+    }
+
+    return configObj;
+  }
+
+  /**
+   * This method provides config object when atleast facilitator config is provided.
+   * @param options It contains path to facilitator and mosaic config.
+   * @returns {Config} Config object encapsulating facilitator and mosaic configs.
+   */
+  private static facilitatorConfigOption(options: any): Config {
+    let configObj = {} as any as Config;
+    // wen no origin and aux chain provided
+    if (options.facilitatorConfig !== undefined) {
+      if (options.mosaicConfig !== undefined) {
+        configObj = Config.getConfigFromPath(options.mosaicConfig, options.facilitatorConfig);
+      } else {
+        const facilitatorConfig: FacilitatorConfig = FacilitatorConfig.fromPath(
+          options.facilitatorConfig,
+        );
+        const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(
+          facilitatorConfig.originChainId,
+        );
+        configObj = new Config(mosaicConfig, facilitatorConfig);
+      }
+    }
+    return configObj;
+  }
+
+  /**
+   * It verifies chain ids in facilitator config.
+   * @param {FacilitatorConfig} facilitatorConfig Facilitator object containing facilitator config.
+   * @param {string} originChain Chain name of the origin chain.
+   * @param {string} auxChain Chain id of the auxiliary chain.
+   * @returns {boolean} `true` if chain ids are present in mosaic config.
+   */
+  private static verifyChainIdInFacilitatorConfig(
```

Comment:
Since the function returns boolean, it's naming can be improved. 
```suggestion
  private static isValidChainIdInFacilitatorConfig(
```

Reply:
It no more returns boolean, so kept the same name.